### ISSUE TITLE
Get week start and end days

### ISSUE THREAD

**Issue by sestrella** [chronotope/chrono#654]
Inspired by Rails [DateAndTime::Calculations](https://edgeapi.rubyonrails.org/classes/DateAndTime/Calculations.html) module, it would be useful to have a set of functions that return week boundaries based on a weekday. This proposal - if approved - intends to add the following functions to `IsoWeek` implementation:

```rs
fn start_day(&self, weekday: Weekday) -> NaiveDate {
  NaiveDate::from_isoywd(self.year(), self.week(), weekday)
}
```

Equivalent to Rails [beginning_of_week](https://edgeapi.rubyonrails.org/classes/DateAndTime/Calculations.html#method-i-beginning_of_week).

```rs
fn end_day(&self, weekday: Weekday) -> NaiveDate {
  self.start_day(weekday) + Duration::days(6)
}
```

Equivalent to Rails [end_of_week](https://edgeapi.rubyonrails.org/classes/DateAndTime/Calculations.html#method-i-end_of_week).

Additionally, a function that returns the start and end of the week at once could be added as well:

```rs
fn start_and_end_days(&self, weekday: Weekday) -> (NaiveDate, NaiveDate) {
  let start_day = self.start_day(weekday);
  (start_day, start_day + Duration::days(6))
}
```

**Usage**

A common use case would be getting the week boundaries based on a day:

```rs
let week = Local::today().iso_week();

let start_day = week.start_day(Weekday::Mon)
let end_day = week.end_day(Weekday::Mon)
// or
let (start_day, end_day) = week.start_and_end_days(Weekday::Mon)
```

**Comment by djc** [comment]
Can you talk more about the use case, what you are trying to achieve that is harder with the current API than (you think) it should be? There are any number of ways we could go about this, but you're describing potential solutions more than the problem you started out with.

**Comment by sestrella** [comment]
Sure let me provide more context, when it comes down to reports it is quite common to collect data within a date range such as weekly, monthly, quarterly, or yearly, therefore, getting week boundaries based on a specific date could be useful, in fact, that is the main reason why I came out with this proposal since right now I'm pulling data from an API that takes `start_day` and `end_day` as arguments. About

> what you are trying to achieve that is harder with the current API than

I would say the current API is very polished and coming up with a solution for the use case described above was not difficult, but at the same time, it feels like a common use case which is why I made this proposal. I hope this extra context provides more insights about the problem. Thank you for reviewing this proposal.
